### PR TITLE
#427: optimize nightly builds

### PR DESCRIPTION
**Pull request by jan-vcapgemini**
Fixes: #427 

Implements:
* new job management with shared results (documentation and native images)
* dynamic profile builds with automatic archive and upload/download of github artifacts
* update of `actions/upload/download-artifact` to `v4`
* some documentation
* update of `native-maven-plugin` to `0.10.2`
* still some redundancies in the `cli/pom.xml`

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -21,23 +47,26 @@ jobs:
...
+      - name: Build native image
         shell: bash
         run: |
-          maven_config="$(cat .mvn/maven.config)"
-          current_version="${maven_config/#*-Drevision=}"
-          current_version="${current_version/ */}"
           cd cli
-          mvn -B -ntp -Pnative -DskipTests=true package
-      - uses: actions/upload-artifact@v3
+          mvn -B -ntp -P${{ matrix.os }} '-P!build-documentation' -DskipTests=true package
```
Comment (hohwille) [inline .github/workflows/release.yml:59]: I would not create a profile for each matrix OS. Your 4 profiles for this in the POM are 98% copy&paste.
Simply keep a single `native` profile and pass the assembly XML file name part as variable:
```suggestion
          mvn -B -ntp -Dideasy.assembly.id=${{ matrix.os }} -DskipTests=true package
```

File: cli/pom.xml [hunk 2]
```diff
@@ -140,54 +140,141 @@
...
-                      <file>${project.build.directory}/${project.artifactId}-osx-aarch_64.tar.gz</file>
-                      <type>tar.gz</type>
-                      <classifier>mac-arm</classifier>
-                    </artifact>
-                  </artifacts>
+                  <appendAssemblyId>true</appendAssemblyId>
+                  <descriptors>
+                    <descriptor>/src/main/assembly/release-windows-x86_64.xml</descriptor>
+                  </descriptors>
+                </configuration>
```
Comment (hohwille) [inline cli/pom.xml:203]: As described above. Just keep one of these 4 profiles as `native` and simply reference descriptor as
```
<descriptor>/src/main/assembly/release-${ideasy.assembly.id}.xml</descriptor>
```
Rename the descriptors to match the github OS image names and we can avoid all the copy & paste.

**Review by hohwille — APPROVED** [review]
@jan-vcapgemini thanks for this PR. You did a great job finding a good solution for the build chaining. 👍 
However, I left comments for further improvement.
To build the documentation, we can simply build in `documentation` folder.
To build the package, we already build in `cli` folder - hence, no profile to include or exclude build of documentation is required.
We can avoid all the copy&paste of profiles and simplify as commented.

File: cli/pom.xml (hunk 2 above)
Comment (jan-vcapgemini) [reply cli/pom.xml]: I've implemented all of your suggestions now and tested the release and nightly build workflows on my own fork.
Seems to be working as expected.

**Review by hohwille — COMMENTED** [review]
@jan-vcapgemini thanks for the rework. 🚀 
I still have doubts that the documentation is just uploaded and downloaded but not installed into maven local repo and assume this will fail at least for release build but we will see today as I will build a release after merging and maybe this will also proof me wrong.
Ready for merge.

**Comment by hohwille** [comment]
> I still have doubts that the documentation is just uploaded and downloaded but not installed into maven local repo and assume this will fail at least for release build but we will see today as I will build a release after merging and maybe this will also proof me wrong.

In my last review now I finally got it. We do not use dependency-plugin anymore to get the doc PDF from local repo and copy it to `target/package` but this is directly done via upload and download of artifacts. Makes sense now.

**Pull request merged**